src/app/auth/custom-input-field/custom-input-field.component.spec.ts

```diff
@@ -0,0 +1,85 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { FormControl, ReactiveFormsModule, Validators } from '@angular/forms';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatInputModule } from '@angular/material/input';
+import { CustomInputFieldComponent } from './custom-input-field.component';
+
+describe('CustomInputFieldComponent', () => {
+  let component: CustomInputFieldComponent;
+  let fixture: ComponentFixture<CustomInputFieldComponent>;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [CustomInputFieldComponent, ReactiveFormsModule, MatFormFieldModule, MatInputModule]
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(CustomInputFieldComponent);
+    component = fixture.componentInstance;
+  });
+
+  const initializeComponent = (
+    label: string,
+    placeholder: string,
+    control: FormControl
+  ) => {
+    component.label = label;
+    component.placeholder = placeholder;
+    component.control = control;
+    fixture.detectChanges();
+  };
+
+  const checkErrorMessage = (error: any, expectedMessage: string) => {
```

Comment:
You could use the ValidationsErrors type from @angular/forms since it will be a validation error instead of the any type.